compiler/hash-typecheck/src/new/environment/tc_env.rs

```diff
@@ -55,6 +55,7 @@ tc_env! {
     #hide env: Env<'tc>,
     diagnostics: DiagnosticsStore,
     ast_info: AstInfo,
+    primitives_or_unset: DefinedPrimitivesOrUnset,
```

Comment:
What is the reasoning for this naming convention?

Reply:
Well it is an option, so either we have defined the primitives or they have not yet been defined.

Reply:
Surely they are set at the beginning of initialisation?

Reply:
They are set a bit after during resolution. TcEnv needs a default value for it so it is None, and then there is an accessor .primitives() which is basically .primitives_pr_unset().unwrap(). They cannot be set at the very beginning because creating the primitives requires access to TcEnv, which it is also part of.